prime-router/src/main/kotlin/azure/CheckFunction.kt

```diff
@@ -107,19 +134,46 @@ class CheckFunction : Logging {
     /**
      * Any normal return is success.  Any exception thrown is failure.
      */
-    fun testSftp(sftpTransportType: SFTPTransportType, receiver: Receiver, responseBody: MutableList<String>) {
+    fun testSftp(sftpTransportType: SFTPTransportType, receiver: Receiver, sftpFile: SftpFile?, responseBody: MutableList<String>) {
         val host = sftpTransportType.host
         val port = sftpTransportType.port
         val path = sftpTransportType.filePath
+        logger.info("SFTP Transport $sftpTransportType")
+        responseBody.add("${receiver.fullName}: SFTP Transport: $sftpTransportType")
         val credential = SftpTransport.lookupCredentials(receiver.fullName)
-        val sshClient = SftpTransport.connect(host, port, credential)
-        responseBody.add("${receiver.fullName}: Able to Connect to sftp site.  Now trying an `ls`...")
+        var sshClient = SftpTransport.connect(host, port, credential)
+        responseBody.add("${receiver.fullName}: Able to Connect to sftp site")
+        sftpFile?. let {
+            logger.info("Attempting to upload ${it.name} to $sftpTransportType")
+            SftpTransport.uploadFile(sshClient, path, it.name, it.contents.toByteArray())
+            responseBody.add("${receiver.fullName}: Uploaded file '${sftpFile.name}' to SFTP transport")
```

Comment:
What happens if a file with that name already exists? Does `uploadFile` error out, or does it overwrite? Can we just make it error constantly? I don't want to risk ever overwriting valid data.

Reply:
Will do. I was relying on the UUID to hopefully create a unique file name but no guarantee and it should check. I'll see how uploadFile handles this situation and either check to see if it exists and regenerate the file name or just throw an exception.

Reply:
I know the odds of UUID collision is practically nil, but it's definitely better to fail and be safe just in case the stars do align unexpectedly.

Reply:
Verified that uploading an existing file does overwrite the file that is there, so I added some precautions with this commit:
https://github.com/CDCgov/prime-data-hub/pull/1115/commits/a0182a263cf9b6488526a7d3ba6934ecb68b9fe6

Summary of those changes:
- overloaded the `ls()` method in the SftpTransport with a signature that accepts a RemoteResourceFilter
- added a check using the filter above and throw an exception if the file already exists
- changed the upload check to use the filter vs iterating through the bulk ls return (seems more accurate)